ui/packages/components/src/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,117 @@
+import { useEffect, useState } from 'react';
+
+import { Button } from '../Button';
+import { Popover, PopoverClose, PopoverContent, PopoverTrigger } from '../Popover';
+import { Switch, SwitchLabel, SwitchWrapper } from '../Switch';
+import { combineDayAndTime, formatDayString, formatTimeString } from '../utils/date';
+import { Calendar } from './Calendar';
+import { DateInputButton } from './DateInputButton';
+import { TimeInput } from './TimeInput';
+
+type DatePickerProps = {
+  datePickerDate?: Date;
+  setDatePickerDate: React.Dispatch<React.SetStateAction<Date | undefined>>;
+};
+
+export function DatePicker({ datePickerDate, setDatePickerDate }: DatePickerProps) {
+  const [buttonCopy, setButtonCopy] = useState<string>(
+    datePickerDate?.toISOString() || 'Enter Date and Time'
+  );
+  const [selectedDay, setSelectedDay] = useState<Date | undefined>(datePickerDate);
+  const [selectedTime, setSelectedTime] = useState<Date | undefined>(datePickerDate);
+  const [calendarOpen, setCalendarOpen] = useState(false);
+  const [is24HourFormat, setIs24HourFormat] = useState(false);
+  const [dayString, setDayString] = useState<string>('');
+  const [timeString, setTimeString] = useState<string>('');
+  const [isValidTime, setIsValidTime] = useState(true);
+
+  useEffect(() => {
+    // Reset selected day and time when the popover is closed
+    if (!calendarOpen) {
+      setSelectedDay(datePickerDate);
+      setSelectedTime(datePickerDate);
+      setIsValidTime(true);
```

Comment:
This is causing the datepicker to forget the existing value when it's reopened:

https://github.com/inngest/inngest/assets/20100586/14fc6a77-f51f-4f5a-89ce-cd895d4239cb